compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1788,7 +1788,23 @@ impl<'a> Parser<'a> {
                     }
                 }
             } else {
-                self.expected_ident_found()
+                let mut err = self.expected_ident_found();
+                if let Some((ident, _)) = self.token.ident() && ident.as_str() == "let" {
+                    self.bump(); // `let`
+                    let span = self.prev_token.span.until(self.token.span);
+                    err.span_suggestion(
+                        span,
+                        "remove the let, the `let` keyword is not allowed in struct field definitions",
+                        String::new(),
+                        Applicability::MachineApplicable,
+                    );
+                    err.span_note(span, "the `let` keyword is not allowed");
+                    err.span_note(span, "see <https://doc.rust-lang.org/book/ch05-01-defining-structs.html> for more information");
```

Comment:
Lets make these notes instead of span notes. If we look at the cli output, we are pointing at the `let` four times now, one in the main error, two in the notes and one in the suggestion.

```suggestion
                    err.note(span, "the `let` keyword is not allowed in `struct` fields");
                    err.note(span, "see <https://doc.rust-lang.org/book/ch05-01-defining-structs.html> for more information");
```

I'm not opposed to increasing verbosity when it adds more information, but in this case using notes the output would be


```
error: expected identifier, found keyword `let`
  --> $DIR/removed-syntax-field-let.rs:2:5
   |
LL |     let foo: (),
   |     ^^^ expected identifier, found keyword
   |
  = note: the `let` keyword is not allowed in `struct` fields
  = note: see <https://doc.rust-lang.org/book/ch05-01-defining-structs.html> for more information
help: remove the let, the `let` keyword is not allowed in struct field definitions
   |
LL -     let foo: (),
LL +     foo: (),
   |
```

which has better information density and personally think it doesn't look half bad. The only issue is the difference between `  = note:` and `help:` clashing slightly in the output, but that's how the formatting operates. If you believe we should improve it (and I'm not against that either!) we should tackle that in the diagnostic formatting logic, not in a specific diagnostic. Does this make sense?

r=me after addressing the nitpick and squashing the commits.